.ci/pitest.groovy

```diff
@@ -0,0 +1,302 @@
+import groovy.io.FileType
+import groovy.transform.Field
+import groovy.xml.XmlParser
+import groovy.xml.XmlUtil
+
+@Field final static String SEPARATOR = System.getProperty("file.separator")
+
+@Field final static Set<String> PROFILES = new HashSet<>(Arrays.asList(
+        "pitest-annotation", "pitest-design", "pitest-metrics",
+        "pitest-modifier", "pitest-naming", "pitest-sizes",
+        "pitest-whitespace", "pitest-api", "pitest-blocks",
+        "pitest-packagenamesloader", "pitest-common-2", "pitest-misc",
+        "pitest-xpath", "pitest-filters", "pitest-regexp",
+        "pitest-meta", "pitest-tree-walker", "pitest-utils",
+        "pitest-java-ast-visitor", "pitest-main", "pitest-header",
+        "pitest-imports", "pitest-common", "pitest-ant", "pitest-indentation",
+        "pitest-javadoc", "pitest-coding-1", "pitest-coding-2",
+        "pitest-coding-require-this-check"
+))
```

Comment:
Let's make this one profile per line, for less diff later.

Reply:
Good point, do you propose that we pull them from pom.xml?

Reply:
Well I was thinking why couldn't we pull them from our `pitest.sh --list` or integrating that process in this groovy script. If we need to we can duplicate the same code, but I would hope we don't need to have 2 versions of the code doing the same thing.

Reply:
This script is a complete replacement for pitest.sh, and already has `--list` functionality

Reply:
Then yes, this should go against the POM like the script does. I did not realize this is a replacement to `pitest.sh`.

Reply:
Then let's recreate `--list` implementation from pitest.sh here, and use this to build the set. Then we never have discrepancies.

Reply:
Done.